src/cloud-api-adaptor/pkg/userdata/provision.go

```diff
@@ -149,60 +162,102 @@ func parseUserData(userData []byte) (*CloudConfig, error) {
 	return &cc, nil
 }
 
-func findConfigEntry(path string, cc *CloudConfig) []byte {
-	for _, wf := range cc.WriteFiles {
-		if wf.Path != path {
-			continue
-		}
-		return []byte(wf.Content)
+func writeFile(path string, bytes []byte) error {
+	// Ensure the parent directory exists
+	err := os.MkdirAll(filepath.Dir(path), 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create directory: %w", err)
 	}
+
+	err = os.WriteFile(path, bytes, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to write file %s: %w", path, err)
+	}
+	logger.Printf("Wrote %s\n", path)
 	return nil
 }
 
-type entry struct {
-	path     string
-	optional bool
+func isAllowed(path string, filesList []string) bool {
+	for _, listedFile := range filesList {
+		if listedFile == path {
+			return true
+		}
+		if strings.HasSuffix(path, listedFile) {
```

Comment:
is there a reason why we don't check for the full path, but only the suffix? Afaik we hardcode the pathes when generating the `write_files` directive, so we could also use the full path here.

Reply:
Agreed, we can match the full path.